studio/components/interfaces/TableGridEditor/SidePanelEditor/ForeignKeySelector/ForeignKeySelector.utils.tsx

```diff
@@ -0,0 +1,46 @@
+import { FOREIGN_KEY_DELETION_ACTION } from 'data/database/database-query-constants'
+
+export const generateDeletionActionDescription = (deletionAction: string, reference: string) => {
+  switch (deletionAction) {
+    case FOREIGN_KEY_DELETION_ACTION.NO_ACTION:
+      return (
+        <>
+          Deleting a record from <code className="text-xs text-scale-1100">{reference}</code> will{' '}
+          <span className="text-brand-900 opacity-75">raise an error</span> if there exists
+          referencing rows from this table
```

Comment:
"if there exists referencing rows from this table" -> "if there are records existing in this table that reference it"

Reply:
Also think think the text should be underlined here instead of green, as it looks a bit like a link.

Reply:
hummm it wasn't super clear to me that this text below was contextual. The dropdown covered the text, so I didn't realize it changed based on the dropdown. I did realize eventually ....but maybe worth a bigger callout, and repeat the heading?

<img width="468" alt="CleanShot 2023-03-07 at 12 33 23@2x" src="https://user-images.githubusercontent.com/105593/223478232-18846767-144b-46cd-b8f4-20288c59d805.png">

Reply:
hmm i'm not inclined to render a call out here as it takes up quite a bit of real-estate + attention
in the future we'll also support upgrade action for foreign keys as well which will be a similar UI to this

but i do agree that the description is not that visible cause the dropdown is hiding it so the change in desc is not apparent

Reply:
im thinking we stick to using the description text here but 
- show the deletion action again as suggested
- highlight the consequence with amber instead of green

![image](https://user-images.githubusercontent.com/19742402/223639754-521529cd-5112-493a-8e62-1e1ba0b31822.png)